pymeasure/instruments/lakeshore/lakeshore_base.py

```diff
@@ -0,0 +1,117 @@
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+import numpy as np
+from time import sleep, time
+
+from pymeasure.instruments import Instrument, Channel
+from pymeasure.instruments.validators import strict_discrete_set
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class LakeShoreTemperatureInputChannel(Channel):
+    """ Temperature input channel on a lakeshore temperature monitor. Reads the temperature in
+    kelvin, celcius, or sensor units. Also provides a method to block the program until a given
+    stable temperature is reached.
+    """
+
+    kelvin = Instrument.measurement(
+        'KRDG? {ch}',
+        """Read the temperature in kelvin from a channel."""
+    )
+    celcius = Instrument.measurement(
+        'CRDG? {ch}',
+        """Read the temperature in celcius from a channel."""
+    )
+    sensor = Instrument.measurement(
+        'SRDG? {ch}',
+        """Read the temperature in sensor units from a channel."""
+    )
+
+    def wait_for_temperature(self, target, unit='kelvin', accuracy=0.1,
+                             interval=1, timeout=360,
+                             should_stop=lambda: False):
+        """ Blocks the program, waiting for the temperature to reach the target
+        within the accuracy (%), checking this each interval time in seconds.
+
+        :param target: Target temperature in kelvin, celcius, or sensor units.
+        :param accuracy: An acceptable percentage deviation between the
+                         target and temperature
+
+        :param timeout: A timeout in seconds after which an exception is raised
+        :param should_stop: A function that returns True if waiting should stop, by
+                            default this always returns False
+        """
+
+        def percent_difference(temperature):
+            return np.abs(100 * (temperature - target) / target)
+
+        t = time()
+        target_acquired = False
+        while not target_acquired:
+            reading = getattr(self, unit)
+            reading = np.array([reading]) if self.id != '0' else np.array(reading)
```

Comment:
Why this `if`? Is this channel of the Lakeshore224 formatting replies differently? If so, might be good to call that out in a comment here. (Also in other docstrings, maybe?)

Reply:
The if is here because for some models, you can use '0' as the channel argument and it returns a list with the values for all channels at once. So instead of having to do queries on each individual channel you can just do a single query on channel '0'.

Since I convert responses to numpy arrays before the `np.all` call, I added this `if` so that when channel 0 was queried you would still end up with `reading` as a 1-dimensional array. Though I am realizing that for the evaluation of `np.all` or `np.allclose` it doesn't really matter, so this if is unnecessary

Reply:
> so that when channel 0 was queried you would still end up with reading as a 1-dimensional array.

A worthy design goal IMO, getting `sometype` or `list-of-sometype` back depending on circumstances feels annoying to me.